### DISCUSSION TITLE
[Search Directory] Toggle files and directories by pressing CTRL-D or CTRL-F

### DISCUSSION THREAD

**Discussion by stevalkr** [PatrickF1/fzf.fish#303] [Q&A] [answered]
with fzf's out-of-the-box fish extension, I can change the source

```
FZF_DEFAULT_COMMAND='find . -type f' \
  fzf --bind 'ctrl-d:reload(find . -type d),ctrl-f:reload(eval "$FZF_DEFAULT_COMMAND")' \
      --height=50% --layout=reverse
```

but I don't find a way to achieve this, what should I replace `eval "$FZF_DEFAULT_COMMAND"` with?

**Comment by PatrickF1** [comment]
Hi Steve, are you asking how to configure [Search Directory](https://github.com/PatrickF1/fzf.fish#-search-directory) so that reloads with different input on Ctrl-D and Ctrl-F?

**Reply by stevalkr** [reply]
Yes, I don't understand how [this function](https://github.com/PatrickF1/fzf.fish/blob/883b78c8d7955b049f670a1e6ac74c0ea3d388ed/functions/_fzf_search_directory.fish#L25) works and failed to configure it. Would be great if you could provide some guides.

**Reply by PatrickF1** [reply, marked as answer]
Try passing those key bindings into `fzf_directory_opts`
Something like
```
set fzf_directory_opts "--bind=ctrl-d:reload(fd --type directory),ctrl-f:reload(fd --type file)"
```
just tested it and it works for me. Make sure you have fd installed

Would you mind marking this as answer if it's what you want?

**Reply by stevalkr** [reply]
Yes it works, but is it possible to use default options of `fzf_directory_opts`? I tried to bind `reload(fd \$fzf_arguments)` but they don't look the same, like colors of symlinks, etc.

**Reply by PatrickF1** [reply]
They are already included automatically through this line: https://github.com/PatrickF1/fzf.fish/blob/883b78c8d7955b049f670a1e6ac74c0ea3d388ed/functions/_fzf_search_directory.fish#L9

**Reply by stevalkr** [reply]
I finally realized it's fd options not fzf options I want 😄 , currently I set

```
set -U fzf_directory_opts "--bind=ctrl-d:reload(fd --type d --type l --color=always --strip-cwd-prefix)" \
                          "--bind=ctrl-f:reload(fd --color=always)"
```

but failed to use `$fd_cmd` as it's not exported to fzf, so there's no `$fzf_fd_opts` and [base directory](https://github.com/PatrickF1/fzf.fish/blob/883b78c8d7955b049f670a1e6ac74c0ea3d388ed/functions/_fzf_search_directory.fish#L18-L22) trick as I understand it

**Reply by PatrickF1** [reply]
Correct, you won't be able to use fd_cmd or fzf_fd_opts because they aren't exported to fzf. I suppose you can try exporting them.
Anyway, would you mind marking one of my responses as the answer so this question can be marked as closed please?

**Reply by stevalkr** [reply]
Thanks!
